### PR TITLE
4786 - Add ids badge component

### PR DESCRIPTION
**Pull request by ericangeles**
**Explain the details for making this change. What existing problem does the pull request solve?**

Implements the IDS Badge for the  web components. The IDS Badge components  used to inform merchants of the status of an object or of an action that’s been taken.

**Related github/jira issue (required)**:

Closes https://github.com/infor-design/enterprise/issues/4786

**Steps necessary to review your pull request (required)**:
- Pull this branch, build, and run the app
- Go to http://localhost:4300/ids-badge
- Review the codes/implementations

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
~~- [ ] A note to the change log.~~

### CODE REVIEW

File: src/ids-badge/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+#Ids Badge Component
+
+## Description
+
+The IDS Badge Compoment consists of a small circle containing alphanumeric or icon that appears to another object.
+
+## Use Cases
+
+Badges can be use to give a non-critical status update on a piece of information or action to your application.
```
Comment (tmcconechy) [inline src/ids-badge/README.md:9]:
```suggestion
- Badges can be use to give a non-critical status update on a piece of information or action to your application.
- Badges can be used to provide users with information about an object that is worthy of attention.
```

File: src/ids-badge/README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+#Ids Badge Component
+
+## Description
+
+The IDS Badge Compoment consists of a small circle containing alphanumeric or icon that appears to another object.
```
Comment (tmcconechy) [inline src/ids-badge/README.md:5]:
```suggestion
The IDS Badge Compoment consists of a small circle containing alphanumeric or icon that appears to another object. Badges are used to display alphanumeric values. A User can easily skim object-related values. 
```

File: src/ids-badge/README.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+## Features (With Code Examples)
+
+A normal/ no properties badge used as a web component.
+
+```html
+<ids-badge>5</ids-badge>
+```
+
+A colored badge is done by adding `color` attribute and one of the following:
```
Comment (tmcconechy) [inline src/ids-badge/README.md:25]:
```suggestion
A colored badge is done by adding a `color` attribute and one of the following:
```

File: src/ids-badge/README.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+- Color
+- Shape
+
+## Keyboard Guidelines
+
+Badges do not have tab stops and have no keyboard interaction on their own, but they may be placed in a grid cell or object that has tab focus.
+
+## Accessibility Guidelines
+
+- 1.4.1 Use of Color - Color is not used as the only visual means of conveying information, indicating an action, prompting a response, or distinguishing a visual element. Ensure the color tags that indicate state like OK, cancel, ect have other ways to indicate that information. This is failing.
```
Comment (tmcconechy) [inline src/ids-badge/README.md:70]:
```suggestion
This is failing.
```
Let's fix that - can we make the green one have dark text instead? I know its a change but better to pass

File: src/ids-badge/README.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+- 1.4.3 Contrast (Minimum) - The visual presentation of text and images of text has a contrast ratio of at least 4.5:1.   Ensure the color tags pass contrast.
```
Comment (tmcconechy) [inline src/ids-badge/README.md:71]:
```suggestion
- 1.4.3 Contrast (Minimum) - The visual presentation of text and images of text has a contrast ratio of at least 4.5:1.   Ensure the color badges pass contrast.
```

File: src/ids-badge/ids-badge.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  constructor() {
+    super();
+  }
+
+  /**
+   * Return the properties we handle as getters/setters
+   * @returns {Array} The properties in an array
+   */
+  static get properties() {
+    return ['color', 'shape'];
```
Comment (tmcconechy) [inline src/ids-badge/ids-badge.js:24]:
```suggestion
    return ['color', 'shape'];
```

Could put this in ids-constants (color exists already)

**Review by tmcconechy — COMMENTED** [review]
Looks good just a couple small comments.

**Review by tmcconechy — APPROVED** [review]

**Review by deep7102 — APPROVED** [review]

**Pull request merged**